user_sync/rules.py

```diff
@@ -543,11 +589,20 @@ def update_dashboard_users_for_connector(self, organization_info, dashboard_conn
 
                 if (manage_groups):
                     # Next, check if that user is in mapped groups and if so, remove from those groups
-                    current_groups = self.normalize_groups(dashboard_user.get('groups'))
-                    groups_to_remove = current_groups & organization_info.get_mapped_groups()
-                    if groups_to_remove != None and len(groups_to_remove) > 0:
+                    dashboard_user_groups = dashboard_user.get('groups')
+                    current_group_names = self.normalize_groups(dashboard_user_groups)
+
+                    # cycle through dashboard group names, and add target groups sharing those names to remove list
+                    groups_to_remove = set()
+                    for current_group_name in current_group_names:
+                        for target_group in organization_info.get_mapped_groups():
```

Comment:
Thinking you should have a routine called organization_info.get_mapped_group_with_name